### PR TITLE
Hook service: useHook and mock-block-dock improvements

### PR DESCRIPTION
**Pull request by CiaranMn**
This PR makes the following improvements following testing of the hook service in the Hub:

**mock-block-dock**

Make toolbar appear on hover, not omnipresent, and improve formatting + styling.

Kapture 2022-10-25 at 09 31 24

Also now doesn't show the toolbar in readonly mode

**@blockprotocol/hook**

Previously, `useHook` was sending a request to tear down the hook before requesting any updates. This meant that the embedding application would tear down the hook and then be asked to update it. 

MBD's implementation of the hook service was returning an error when the hook to update was not found, causing the Header block to crash in the Hub when switching DOM nodes.

This PR stops `useHook` from sending any tear down requests before updating a hook, since #676 made it so that any hook property can be changed in an update request (making it unnecessary to tear down the hook unless the component is being unmounted).

## Next steps

1. Update text blocks to use the new packages
2. Update and make #697 ready to review

### CODE REVIEW

File: packages/@blockprotocol/hook/src/react.ts [hunk 1]
```diff
@@ -243,50 +240,45 @@ export const useHook = <T extends HTMLElement>(
 
       existingHookRef.current = hook;
 
-      teardownPromise
```
Comment (CiaranMn) [inline packages/@blockprotocol/hook/src/react.ts:246]: enable hide whitespace to see what's changed other than removing this teardown

File: packages/@blockprotocol/hook/src/react.ts [hunk 2]
```diff
@@ -243,50 +240,45 @@ export const useHook = <T extends HTMLElement>(
...
+                // eslint-disable-next-line no-console
+                console.error(response.errors);
+                throw new Error("Unknown error in hook");
               }
-            });
-        })
-        .catch((err) => {
-          catchError(() => {
-            throw err;
-          });
```
Comment (nathggns) [inline packages/@blockprotocol/hook/src/react.ts:289]: Blocking: this needs to be here so React catches the errors thrown above – unsure why it's been removed.

File: packages/mock-block-dock/src/hook-portals/text/toolbar.tsx [hunk 3]
```diff
@@ -1,24 +1,77 @@
+import { ReactElement, useEffect, useRef } from "react";
+import { createPortal } from "react-dom";
+import { Editor, Range } from "slate";
+import { useFocused, useSlate } from "slate-react";
+
 import { MarkButton } from "./mark-button";
 
+const Portal = ({ children }: { children: ReactElement }) => {
+  return createPortal(children, document.body);
+};
```
Comment (nathggns) [inline packages/mock-block-dock/src/hook-portals/text/toolbar.tsx:10]: Could you just call `createPortal` in `Toolbar`?

**Review by nathggns — DISMISSED** [review]

File: packages/@blockprotocol/hook/src/react.ts (hunk 2 above)
Comment (CiaranMn) [reply packages/@blockprotocol/hook/src/react.ts]: Thank you, restored in https://github.com/blockprotocol/blockprotocol/pull/707/commits/8cbb862915ec99adce6ca230d209ec5a07b5a8af

File: packages/@blockprotocol/hook/src/react.ts [hunk 4]
```diff
@@ -190,9 +190,6 @@ export const useHook = <T extends HTMLElement>(
 
     const existingHookId = existingHookRef.current?.id;
 
-    const teardownPromise =
-      existingHookRef.current?.teardown?.().catch() ?? Promise.resolve();
-
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply packages/@blockprotocol/hook/src/react.ts]: Something like this? https://github.com/blockprotocol/blockprotocol/pull/707/commits/8cbb862915ec99adce6ca230d209ec5a07b5a8af

File: packages/mock-block-dock/src/hook-portals/text/toolbar.tsx (hunk 3 above)
Comment (CiaranMn) [reply packages/mock-block-dock/src/hook-portals/text/toolbar.tsx]: Yes - https://github.com/blockprotocol/blockprotocol/pull/707/commits/8cbb862915ec99adce6ca230d209ec5a07b5a8af

File: packages/@blockprotocol/hook/src/react.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (CiaranMn) [reply packages/@blockprotocol/hook/src/react.ts]: @nathggns, I've asked @kachkaev to check that I've addressed your feedback and approve on that basis (if he agrees that I have) so we can close out the text tasks. 

If you think that what has been done here needs improvement (and this PR has been merged) I am happy to make further improvements in follow up

[the remark this answers is not in the document]
Comment (CiaranMn) [reply packages/@blockprotocol/hook/src/react.ts]: @kachkaev, the concern here was that we've removed the teardown that cancelled any pending request for the hook, which might mean that a response to an earlier hook mesage was received out of order. I've therefore added in a `cancel` back in this place which cancels any in-flight requests and bails out of the `.then` part of the promise - see check for `(!controller.signal.aborted)`

**Review by kachkaev — APPROVED** [review]

**Pull request merged**